vuu/src/main/scala/org/finos/vuu/net/rpc/RpcMethodHandler.scala

```diff
@@ -0,0 +1,27 @@
+package org.finos.vuu.net.rpc
+
+import org.finos.vuu.net.RequestContext
+
+trait RpcMethodHandler {
+  def call(rpcParams: RpcParams): RpcMethodCallResult
+}
+
+class RpcJavaFunctionMethodHandler(handler: java.util.function.Function[RpcParams, RpcMethodCallResult]) extends RpcMethodHandler {
+  override def call(rpcParams: RpcParams): RpcMethodCallResult = handler(rpcParams)
+}
+
+class RpcFunctionMethodHandler(handler: RpcParams => RpcMethodCallResult) extends RpcMethodHandler {
+  override def call(rpcParams: RpcParams): RpcMethodCallResult = handler(rpcParams)
+}
+
+class RpcParams(val params: Array[Any], val namedParams: Map[String, Any], ctx: RequestContext)
+
+trait RpcMethodCallResult {}
```

Comment:
Can be a separate discussion but looks like this usecase is a good candidate for reviewing the Result Pattern discussion